internal/controller/preservation/preservation_test.go

```diff
@@ -57,7 +57,9 @@ func verifyFn(targetObject types.NamespacedName, expectedObjects []client.Object
 			return true
 		}
 	}
-	Expect(err).To(HaveOccurred(), fmt.Sprintf("Expected error since resource %s/%s should not exist", targetObject.Namespace, targetObject.Name))
+	if err != nil {
+		Expect(err).To(HaveOccurred(), fmt.Sprintf("Expected error since resource %s/%s should not exist", targetObject.Namespace, targetObject.Name))
```

Comment:
Isn't this not really checking anything? If you pass an error it will pass the test, if you don't it won't run the test ... shouldn't we have both a "I'm expecting an error" field and the actual error that potentially occurred?

Reply:
FWIW it seems like previously the logic was "if we didn't find the target object we should have an error, which makes sense to me.

Reply:
Good catch! Updated :)